src/tutorial/play-audio.rst

```diff
@@ -0,0 +1,103 @@
+Play an audio
+=============
+
+Now that you know how to create a context,
+let's get into the most essential use case: playing audio.
+
+Creating a source
+-----------------
+
+To play an audio, you have to create a source. This source
+is an imaginary sound broadcaster, whose positions and properties
+can be changed to create desired effects.
+
+.. code-block:: python
+
+   from palace import Device, Context, Source
+
+   with Device(), Context() as ctx:
+       with Source() as src:
+           # to be written
+
+Just like for the case of `Context`, `Source` creation
+requires a context, but here the context is passed implicitly.
+
+Decode the audio file
+---------------------
+
+Palace has a module level function `decode`, which decodes audio file
+automatically, and this decoded file is a `Decoder` object. This object
+can be played by a simple `play()` method.
+
+.. code-block:: python
+
+   from palace import Device, Context, Source, decode
+
+   fname = 'some_audio.ogg'
+
+   with Device(), Context() as ctx:
+       with Source() as src:
+           dec = decode(fname)
+
+We are almost there. Now, let's look at the document for `Decoder.play()`.
+The method takes 3 parameters: `chunk_len`, `queue_size`, and `source`.
+
+The source object is optional, because if you don't have it, a new source
+will be generated by default.
+
+The audio is divided into chunks, each of which is of length `chunk_len`.
+Then `queue_size` is the number of these chunks that it will play.
+
+.. TODO: I think it's better to include a diagram here. Add later
+
+.. code-block:: python
+
+   from palace import Device, Context, Source, decode
+
+   fname = 'some_audio.ogg'
+
+   with Device(), Context() as ctx:
+       with Source() as src:
+           dec = decode(fname)
+	   dec.play(12000, 4, src)
+
+But we don't want it to play only a small part. of the audio. We want it to
+play all of it. How do we do that? The answer is a loop.
+
+`Context` has a method, `update()`, which update itself and the source.
```

Comment:
ReST uses double tick for fixed width text, however it's a lot better if you can use the `:py:...` like in design principles.  Moreover, please separate sentences by 2 spaces.

Reply:
Ping on the sentence separation thing.